src/irmin-pack/irmin_pack_layers.ml

```diff
@@ -964,34 +921,22 @@ struct
       incr errors;
       Lwt.return_unit
     in
-    let check_tree root =
-      let node k = X.Node.CA.check t.X.Repo.node ~none k in
-      let contents k = X.Contents.CA.check t.X.Repo.contents ~none k in
-      Repo.iter_nodes t ~min:[] ~max:[ root ] ~node ~contents ()
-    in
-    let check_commit commit =
-      let some c = check_tree (X.Commit.Val.node c) in
-      X.Commit.CA.check t.X.Repo.commit ~none ~some commit
```

Comment:
You forgot the check on commit. Adding
` let commit k = X.Commit.CA.check t.X.Repo.commit ~none k in`
back in, and then
`Repo.iter t ~min:[] ~max:hashes ~node ~contents ~commit () >|= fun () ->`
fixes it for me.

Reply:
Good catch! all fixed now